src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,91 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class for loading and persisting data to disk.
+ */
+public class Storage {
+    private static final String saveFilePath = "data/duke.txt";
+
+    /**
+     * Save tasks to disk.
+     * @param tasks ArrayList of tasks to save
+     * @throws IOException if cannot write to saveFilePath
+     */
+    public static void save(ArrayList<Task> tasks) throws IOException {
+        createSaveDirectoryIfNotExists();
+        FileWriter fileWriter = new FileWriter(saveFilePath);
+        for (Task task: tasks) {
+            fileWriter.write(task.toSaveString() + System.lineSeparator());
+        }
+        fileWriter.close();
+    }
+
+    /**
+     * Load tasks from saveFilePath.
+     * @return ArrayList of loaded tasks
+     */
+    public static ArrayList<Task> load() {
+        try {
+            createSaveDirectoryIfNotExists();
+            File file = new File(saveFilePath);
+            Scanner scanner = new Scanner(file);
+            ArrayList<Task> tasks = new ArrayList<>();
+            while (scanner.hasNext()) {
+                String[] components = scanner.nextLine().split("\\|");
+                switch (components[0]) {
+                    case "T":
+                        tasks.add(new Todo(components[2], components[1].equals("1")));
```

Comment:
Can consider declaring a constant where such as done = "1" means to improve readability when creating new tasks.

Reply:
Good suggestion